src/QMCWaveFunctions/SpinorSet.cpp

```diff
@@ -157,45 +157,17 @@ void SpinorSet::mw_evaluateVGLWithSpin(const RefVectorWithLeader<SPOSet>& spo_li
   auto& up_spo_leader             = up_spo_list.getLeader();
   auto& dn_spo_leader             = dn_spo_list.getLeader();
 
-  std::vector<ValueVector> mw_up_psi_work, mw_dn_psi_work;
-  std::vector<GradVector> mw_up_dpsi_work, mw_dn_dpsi_work;
-  std::vector<ValueVector> mw_up_d2psi_work, mw_dn_d2psi_work;
-  mw_up_psi_work.reserve(nw);
-  mw_up_dpsi_work.reserve(nw);
-  mw_up_d2psi_work.reserve(nw);
-  mw_dn_psi_work.reserve(nw);
-  mw_dn_dpsi_work.reserve(nw);
-  mw_dn_d2psi_work.reserve(nw);
-
-  RefVector<ValueVector> up_psi_v_list, dn_psi_v_list;
-  RefVector<GradVector> up_dpsi_v_list, dn_dpsi_v_list;
-  RefVector<ValueVector> up_d2psi_v_list, dn_d2psi_v_list;
-  up_psi_v_list.reserve(nw);
-  up_dpsi_v_list.reserve(nw);
-  up_d2psi_v_list.reserve(nw);
-  dn_psi_v_list.reserve(nw);
-  dn_dpsi_v_list.reserve(nw);
-  dn_d2psi_v_list.reserve(nw);
-
-  ValueVector tmp_val_vec(OrbitalSetSize);
-  GradVector tmp_grad_vec(OrbitalSetSize);
-  for (int iw = 0; iw < nw; iw++)
-  {
-    mw_up_psi_work.emplace_back(tmp_val_vec);
-    up_psi_v_list.emplace_back(mw_up_psi_work.back());
-    mw_dn_psi_work.emplace_back(tmp_val_vec);
-    dn_psi_v_list.emplace_back(mw_dn_psi_work.back());
-
-    mw_up_dpsi_work.emplace_back(tmp_grad_vec);
-    up_dpsi_v_list.emplace_back(mw_up_dpsi_work.back());
-    mw_dn_dpsi_work.emplace_back(tmp_grad_vec);
-    dn_dpsi_v_list.emplace_back(mw_dn_dpsi_work.back());
-
-    mw_up_d2psi_work.emplace_back(tmp_val_vec);
-    up_d2psi_v_list.emplace_back(mw_up_d2psi_work.back());
-    mw_dn_d2psi_work.emplace_back(tmp_val_vec);
-    dn_d2psi_v_list.emplace_back(mw_dn_d2psi_work.back());
-  }
+  ValueVector sized_val(OrbitalSetSize);
+  GradVector sized_grad(OrbitalSetSize);
+  std::vector<ValueVector> mw_up_psi_work(nw, sized_val), mw_dn_psi_work(nw, sized_val);
+  std::vector<GradVector> mw_up_dpsi_work(nw, sized_grad), mw_dn_dpsi_work(nw, sized_grad);
+  std::vector<ValueVector> mw_up_d2psi_work(nw, sized_val), mw_dn_d2psi_work(nw, sized_val);
```

Comment:
It is bad to create and destroy the work space. The per-walker vector can be collected.

Reply:
fixed to collect from per-walker work arrays that already existed